### PR TITLE
Impl user lookup

### LINKED ISSUE #53: User lookup

### PR DESCRIPTION
**Pull request by mmmommm**
close #53

### CODE REVIEW

File: user.go [hunk 1]
```diff
@@ -98,10 +98,37 @@ type UserWithheld struct {
...
+	Errors   []*APIResponseError `json:"errors,omitempty"`
+}
+
+type UserLookUpByUserNameResponse struct {
+	User     *User               `json:"data"`
+	Includes *UserIncludes       `json:"includes,omitempty"`
+	Errors   []*APIResponseError `json:"errors,omitempty"`
+	Title    string              `json:"title,omitempty"`
+	Detail   string              `json:"detail,omitempty"`
+	Type     string              `json:"type,omitempty"`
```
Comment (sivchari) [inline user.go:124]: In what cases is this field needed?

File: user.go [hunk 2]
```diff
@@ -98,10 +98,37 @@ type UserWithheld struct {
...
+}
+
+type UsersLookUpByUserNamesResponse struct {
+	Users    []*User             `json:"data"`
+	Includes *UserIncludes       `json:"includes,omitempty"`
+	Errors   []*APIResponseError `json:"errors,omitempty"`
+	Title    string              `json:"title,omitempty"`
+	Detail   string              `json:"detail,omitempty"`
+	Type     string              `json:"type,omitempty"`
```
Comment (sivchari) [inline user.go:133]: Same reason.

File: user_lookup.go [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+			APIName: "user lookup by username",
+			Status:  resp.Status,
+			URL:     req.URL.String(),
+		}
+	}
+
+	return &user, nil
+}
+
+// lookup multiple users by user names
```
Comment (sivchari) [inline user_lookup.go:161]:
```suggestion
// lookUpUsersByUserNames look up multiple users by user names
```

File: user_lookup.go [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+			APIName: "user lookup by id",
+			Status:  resp.Status,
+			URL:     req.URL.String(),
+		}
+	}
+
+	return &user, nil
+}
+
+// lookup single user by username
```
Comment (sivchari) [inline user_lookup.go:114]:
```suggestion
// lookUpUserByUserName look up single user by username
```

**Review by sivchari — CHANGES_REQUESTED** [review]

File: user.go (hunk 1 above)
Comment (mmmommm) [reply user.go:124]: @sivchari 
when `lookUpUserByUserName` and `lookUpUsersByUserNames` function return 404 error these field are needed.
I've commented on the part of the test case where the field is used, so please have a look there.

File: user_lookup_test.go [hunk 5]
```diff
@@ -0,0 +1,962 @@
...
+				User: nil,
+				Errors: []*gotwtr.APIResponseError{
+					{
+						Parameters: gotwtr.Parameter{
+							UserName: []string{"aaaaaaaaaaaaaaaaaa"},
+						},
+						Message: "The username query parameter value [aaaaaaaaaaaaaaaaaa] does not match ^[A-Za-z0-9_]{1,15}$",
+					},
+				},
+				Title:  "Invalid Request",
```
Comment (mmmommm) [inline user_lookup_test.go:716]:
```
Title    string              `json:"title,omitempty"`
Detail   string              `json:"detail,omitempty"`
Type     string              `json:"type,omitempty"`
```
Here's where the field is being used.

File: user_lookup_test.go [hunk 6]
```diff
@@ -0,0 +1,962 @@
...
+										"aaaaaaaaaaaaaaaaaa"
+									]
+								},
+								"message":"The username query parameter value [aaaaaaaaaaaaaaaaaa] does not match ^[A-Za-z0-9_]{1,15}$"
+							}
+						],
+						"title":"Invalid Request",
+						"detail":"One or more parameters to your request was invalid.",
+						"type":"https://api.twitter.com/2/problems/invalid-request"
+						}
```
Comment (sivchari) [inline user_lookup_test.go]: What is this RBRACE?

File: user_lookup_test.go (hunk 5 above)
Comment (sivchari) [reply user_lookup_test.go:715]: Thanks.

**Review by sivchari — CHANGES_REQUESTED** [review]

**Review by sivchari — APPROVED** [review]

**Pull request merged**